pkg/monitortests/clusterversionoperator/legacycvomonitortests/operators.go

```diff
@@ -35,14 +35,94 @@ func testStableSystemOperatorStateTransitions(events monitorapi.Intervals) []*ju
 			}
 		}
 
-		return "We are not worried about Available=False or Degraded=True blips for stable-system tests yet.", nil
+		// For the non-upgrade case, if any operator has Available=False, fail the test.
+		if condition.Type == configv1.OperatorAvailable {
+			if condition.Status == configv1.ConditionFalse {
+				return "", nil
+			}
+		}
+		return "We are not worried about Degraded=True blips for stable-system tests yet.", nil
 	}
 
 	return testOperatorStateTransitions(events, []configv1.ClusterStatusConditionType{configv1.OperatorAvailable, configv1.OperatorDegraded}, except)
 }
 
+// isInUpgradeWindow determines if the given eventInterval falls within an upgrade window.
+// If we don't find any upgrade starting point, we assume the eventInterval is not in an upgrade window.
+// If we don't find any upgrade ending point, we assume the ending point is at the end of the test.
+func isInUpgradeWindow(eventList monitorapi.Intervals, eventInterval monitorapi.Interval) bool {
+	type upgradeWindowHolder struct {
+		startInterval *monitorapi.Interval
+		endInterval   *monitorapi.Interval
+	}
+
+	var upgradeWindows []*upgradeWindowHolder
```

Comment:
I'll leave this for you to decide, but I think an actual interval for the overall upgrade window using the logic here would be generally useful. That would mean a new minimal monitortest that just creates some new intervals with a new Source like UpgradeWindow and sets the from-to appropriately. Then this function could be scanning all intervals for the UpgradeWindows, see if our interval is within one, and we could chart the overall upgrade start/end times.